dsl_parser/functions.py

```diff
@@ -892,70 +848,113 @@ def handler(v, scope, context, path):
             if not isinstance(func, Function):
                 break
             previous_evaluated_value = evaluated_value
-            evaluated_value = getattr(func, evaluator)(**evaluator_kwargs)
+            evaluated_value = self.evaluate_function(func)
             if scanned and previous_evaluated_value == evaluated_value:
                 break
             scanned = True
         return evaluated_value
-    return handler
+
+    def evaluate_function(self, func):
+        return func.evaluate(self)
 
 
-def plan_evaluation_handler(plan):
-    return _handler('evaluate', plan=plan)
+class _PlanEvaluationHandler(_EvaluationHandler):
+    def __init__(self, plan, runtime_only_evaluation):
+        self._plan = plan
+        self._runtime_only_evaluation = runtime_only_evaluation
 
+    def evaluate_function(self, func):
+        if not self._runtime_only_evaluation and \
+                func.func_eval_type in (HYBRID_FUNC, STATIC_FUNC):
+            return func.evaluate(self)
+        else:
+            if 'operation' in func.context:
+                func.context['operation']['has_intrinsic_functions'] = True
+            return func.raw
+
+    def get_input(self, input_name):
+        try:
+            return self._plan.inputs[input_name]
+        except KeyError:
+            raise exceptions.UnknownInputError(
+                "get_input function references an "
+                "unknown input '{0}'.".format(input_name))
 
-def runtime_evaluation_handler(get_node_instances_method,
-                               get_node_instance_method,
-                               get_node_method,
-                               get_secret_method,
-                               get_capability_method):
-    return _handler('evaluate_runtime',
-                    storage=RuntimeEvaluationStorage(
-                        get_node_instances_method=get_node_instances_method,
-                        get_node_instance_method=get_node_instance_method,
-                        get_node_method=get_node_method,
-                        get_secret_method=get_secret_method,
-                        get_capability_method=get_capability_method))
+    def get_node(self, node_name):
+        for n in self._plan.node_templates:
```

Comment:
maybe move it to list comprehension for and return the first

Reply:
that'd just use more memory unnecessarily (for the list)